java/arcs/core/analysis/ExpressionDataFlowAnalyzer.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+@file:Suppress("UNCHECKED_CAST")
+
+package arcs.core.analysis
+
+import arcs.core.data.Claim
+import arcs.core.data.ParticleSpec
+import arcs.core.data.expression.Expression
+
+private typealias Scope = DependencyNode.AggregateValue
+
+/**
+ * A visitor that parses Paxel [Expression]s to produce data flow dependencies.
+ *
+ * For each [Expression], this visitor produces a [DependencyNode], which can be translated into a
+ * set of [Claim] relationships.
+ *
+ * [DependencyNode]s are DAG structures that help map handle connections in a [ParticleSpec] to the
+ * target [Expression].
+ */
+class ExpressionDataFlowAnalyzer : Expression.Visitor<DependencyNode, Scope> {
```

Comment:
`Expression` is very generic. I would qualify it with `Paxel` somehow. Some suggestions: 
  - `PaxelExpressionAnalyzer`
  - `PaxelDataDependencyAnalyzer`
  - `PaxelDataDependencyExtractor`
  - ...

Reply:
We're in the weird place with "Paxel" name. We pretty much don't use it in the codebase (Expression.kt, EvaluatorParticle.kt, Evaluator.kt, ...) the only exception is PaxelParser.kt

The guideline from internal google codebases to not use codenames in the codebase for naming files and directories, and I think the arguments for that are quite good.

Given the above and given that all of paxel is in arcs/core/data/expression, I think using Expression here is the right thing to do. And I would rename PaxelParser to ExpressionParser and for now use `Expression` as a name for Paxel in codebase, however vague the meaning of expression is.

Reply:
I like `ExpressionDependencyAnalyzer`.